x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawl_details_flyout/crawl_details_summary.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+
+import moment from 'moment';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiIconTip,
+  EuiPanel,
+  EuiSpacer,
+  EuiStat,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { CrawlRequestStats } from '../../types';
+
+interface ICrawlerSummaryProps {
+  crawlDepth: number;
+  crawlType: string;
+  crawlerLogsEnabled: boolean | undefined;
+  domainCount: number;
+  stats: CrawlRequestStats | null;
+}
+
+export const CrawlDetailsSummary: React.FC<ICrawlerSummaryProps> = ({
+  crawlDepth,
+  crawlType,
+  crawlerLogsEnabled,
+  domainCount,
+  stats,
+}) => {
+  const duration = () => {
```

Comment:
[Optional]

When there are things in a render function that are potentially computationally expensive, it is ideal to use [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) so that they're only computed once.